### ISSUE TITLE
Check if all PR files are in the same folder

### ISSUE THREAD

**Issue by enen92** [xbmc/addon-check#60]
According to our submissions rules every pull request must refer to only one addon. PRs usually have "[addon.id] version" as the PR title (although in some cases we have to rename them manually or ask the author to do so).
Sometimes there is a mistake of including more than one addon in a pull request or of producing incorrect pull requests which include older commits (for example the result of an incorrect rebase). 

The CI could check if all the files included in the pull request are constrained to a single folder (with the same name as the addon id); hence detecting if the pull request is valid.

All my issues are just possible features/ideas that come into my mind for the GSOC work on the tool. I can later implement some of them if not addressed during GSOC so please do not interpret them as I am demanding work :).

Regards

**Comment by Razzeee** [comment]
It might make sense to only enable this via arg parameters.
So that I still can do `kodi-addon-checker script.test script.test2` locally, but error on `kodi-addon-checker script.test script.test2` --fail-on-multiple-addons

**Comment by mzfr** [comment]
@enen92 For this we can check if upper directory for all the file is same or not right ?
Maybe do something like `os.chdir("..")` for all the files/folders in depth 1 and then see if that directory is same or not.

**Comment by Razzeee** [comment]
I'm not sure about thise, especially if we need to call the github api. It means we add another hard dependency and very closely couple this checker with our repo rules on prs. Not with good rules for coding.

So if we would run this locally with vscode some day it will just tilt untill we do some weird setup. Maybe it's a different concern and would be better of in it's own tool?

**Comment by Rechi** [comment]
add the following to .travis.yml
`- 'if [ "$TRAVIS_PULL_REQUEST" != "false" ]; then if [ "$(git diff --diff-filter=d --name-only HEAD~ | grep / | cut -d / -f1 | sort | uniq | wc -l)" -gt "1" ]; then exit 1; fi fi'`

**Comment by mzfr** [comment]
@Razzeee @enen92 This is done right ?

**Comment by Razzeee** [comment]
No, I haven't updated our travis scripts yet

**Comment by mzfr** [comment]
If you can tell me for what all branch travis scripts are to need to be updated then I can do it :slightly_smiling_face:

**Comment by Razzeee** [comment]
Well we need to add the stuff posted by rechi above and maybe also add the needed travis-buddy command, that I already added as a test on the plugins repo for the krypton branch.

**Comment by Razzeee** [comment]
@Rechi is that code snippet still how it should look?

**Comment by Rechi** [comment]
Yes, but it would mark some of our own PRs in repo-resources as failed (e.g. mass updates to game.controler.*).

**Comment by Razzeee** [comment]
So we might want to not do this I guess.

**Comment by enen92** [comment]
I think we still need this, mass updates to addons are not common and are usually handled only by team members. In most situations 1 PR = 1 addon update. Would it be possible to add a label to a pull request and make travis to ignore the check? Something like the "No-Jenkins" label in Jenkins?

**Comment by Rechi** [comment]
Here is the script which just checks every commit of a PR separately.
```
if [ "$TRAVIS_PULL_REQUEST" != "false" ]; then
        git rev-list --no-merges --reverse HEAD~..HEAD | while read rev; do
                if [ "$(git diff --diff-filter=d --name-only $rev~..$rev | grep / | cut -d / -f1 | sort | uniq | wc -l)" -gt "1" ]; then
                        echo 'The following commit changes more then one add-on'
                        git --no-pager show --stat $rev
                        exit 1
                fi
        done
fi
```

**Comment by mzfr** [comment]
@Rechi This script is all we need to add in the `.travis.yml` file right? or is there any other changes required to fix the issue?
